src/bindings/swig/ruby/kdb.i

```diff
@@ -47,8 +47,10 @@ namespace std {
 %{
   extern "C" {
     #include "kdbconfig.h"
-    #include "kdbprivate.h" /* required for KEYSET_SIZE */
-    #include "kdblogger.h"
+    /* #include "kdbprivate.h" required for KEYSET_SIZE */
```

Comment:
Why not simply always pass RARRAY_LEN(argv[0]) to the constructor? The ksNew implementation  already makes sure that at least KEYSET_SIZE elements are allocated.

Then KEYSET_SIZE would not be needed.

Otherwise I think this is ready to go. Could you please rebase, add something to the release notes, and add "ready to merge"?

Thank you for working on this!